core/comp_ui.py

```diff
@@ -32,18 +32,23 @@
 def _PromptLen(prompt_str):
   """Ignore all characters between \x01 and \x02 and handle unicode characters.
   In particular, the display width of a string may be different from either the
-  number of bytes or the number of unicode characters."""
+  number of bytes or the number of unicode characters.
+  Additionally, if there are multiple lines in the prompt, only give the length of the last line."""
   escaped = False
   display_str = ""
-  for c in prompt_str:
+  last_line = prompt_str.split('\n')[-1]
```

Comment:
OK now I get your question more... Although I was sort of agnostic about this, it does seem better to strip `\x01` first, and then take the last line of `display_str`?  

Because otherwise you will count stuff that the user intends for you to ignore.   My interpretation of `x01` is "ignore everything inside here when calculating the length".

Reply:
I guess if `\x01` is wrong, that's a user error ... It might nice to give a warning though. Say you have `PS1=$'abc\x01def\n\033[00m\x02> '`, then the prompt will look like it has 2 characters (`> `) but we'll calculate it as having 5 (`abc> `). It will be wrong either way we calculate it (splitting the line first won't ignore the ANSI escapes)